toolchain/sem_ir/formatter.cpp

```diff
@@ -665,7 +665,17 @@ class Formatter {
   template <typename NodeT>
   auto FormatInstructionRHS(NodeT node) -> void {
     // By default, an instruction has a comma-separated argument list.
-    std::apply([&](auto... args) { FormatArgs(args...); }, node.args_tuple());
+    using Info = TypedNodeArgsInfo<NodeT>;
```

Comment:
Perhaps we can put more functionality into `TypedNodeArgsInfo` so this code doesn't need to deal with `StructReflection` and tuples? Could we make it so you write `Info::GetField0(node)` instead of
```
auto tuple = StructReflection::AsTuple(node);
std::get<Info::Arg0Field>(tuple)
```

Reply:
It feels a bit wasteful to me to convert the struct to a tuple once for each field you read, but I suppose that'll all get optimized away. Done.